src/apply.js

```diff
@@ -701,17 +719,20 @@ function processStyle(glStyle, map, styleUrl, options) {
             glLayer.layout ? glLayer.layout.visibility !== 'none' : true
           );
           const functionCache = {};
-          view.on(
-            'change:resolution',
-            updateRasterLayerProperties.bind(
-              this,
-              glLayer,
-              layer,
-              view,
-              functionCache
-            )
-          );
-          updateRasterLayerProperties(glLayer, layer, view, functionCache);
+          // ??? A group wouldn't have a reference to map/view?
+          if (view) {
```

Comment:
This could be solved by listening to the layer's prerender event instead of the view's `change:resolution` event. That event receives the resolution as `event.frameState.viewState.resolution`, which would allow calling `updateRasterLayerProperties()` only when the resolution has changed.

Reply:
Think I have this ready, but noticed that it seems background can't be applied to GeoJSON layer due to the following `updateStyle` function not accounting for it. This implies background can only be applied to `VectorTile` or `Map`? Is that by design or should I add it to `updateStyle`?
```
    const zoom =
      typeof mapOrLayer.getSource === 'function'
        ? mapOrLayer.getSource().getTileGrid().getZForResolution(resolution)
        : mapOrLayer.getView().getZoom();
```